sirepo/template/radia.py

```diff
@@ -1233,29 +1231,15 @@ def _save_fm_sdds(name, vectors, scipy_rotation, path):
     return path
 
 
-def _save_kick_map_sdds(name, x_vals, y_vals, h_vals, v_vals, path):
+def _save_kick_map_sdds(name, path, km_data):
     s = _get_sdds(_KICK_MAP_COLS, _KICK_MAP_UNITS)
     s.setDescription(f"Kick Map for {name}", "x(m), y(m), h(T2m2), v(T2m2)")
-    col_data = []
-    x = []
-    y = []
-    h = []
-    v = []
-    # TODO: better way to do this...
-    for i in range(len(x_vals)):
-        for j in range(len(x_vals)):
-            x.append(0.001 * x_vals[j])
-    for i in range(len(y_vals)):
-        for j in range(len(y_vals)):
-            y.append(0.001 * y_vals[i])
-    for i in range(len(x_vals)):
-        for j in range(len(y_vals)):
-            h.append(h_vals[i][j])
-            v.append(v_vals[i][j])
-    col_data.append([x])
-    col_data.append([y])
-    col_data.append([h])
-    col_data.append([v])
+    col_data = [
+        [numpy.tile(0.001 * numpy.array(km_data.x), len(km_data.x)).tolist()],
```

Comment:
maybe put 0.001 in a constant with a name that clarifies it's role?

Reply:
Good call, [86bc328](https://github.com/radiasoft/sirepo/pull/4785/commits/86bc3289d0fd30f6855bef62135ad38a493cf383)